### PR TITLE
Feature/flow specification

### PR DESCRIPTION
**Pull request by sergeysenja1992**

### CODE REVIEW

File: xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/service/CodeSnippetService.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import static com.icthh.xm.commons.config.client.utils.Utils.nullSafeList;
+import static com.icthh.xm.commons.config.client.utils.Utils.nullSafeMap;
+import static com.icthh.xm.commons.flow.service.CodeSnippetExecutor.SNIPPET;
+import static java.util.stream.Collectors.toList;
+
+@Component
+@LepService(group = "flow")
+public class CodeSnippetService {
+
+    public static final String $$ = "$$";
```
Comment (amedvedchuk) [inline xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/service/CodeSnippetService.java:25]: propose to use alfanumeric name for constant

Comment (sergeysenja1992) [reply xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/service/CodeSnippetService.java:25]: 1 - what name do you suggest?
2 - line 58 will be so long

File: xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/service/FlowService.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    private void assertExits(String flowKey) {
+        if (flowConfigService.getFlow(flowKey) == null) {
+            throw new BusinessException("error.flow.not.found", "Resource with key " + flowKey + " not found");
+        }
+    }
+
+    private void updateConfigurations(List<Configuration> configurations) {
+        log.debug("Updated configs: {}", configurations);
+        log.info("Updated configs.size: {}", configurations.size());
```
Comment (amedvedchuk) [inline xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/service/FlowService.java:116]: propose to change to "Updated configs.count: {}"

File: xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/service/CodeSnippetService.java (hunk 1 above)
Comment (amedvedchuk) [reply xm-commons-flow/src/main/java/com/icthh/xm/commons/flow/service/CodeSnippetService.java:25]: 1 - SEGMENT_SEP

**Review by amedvedchuk — APPROVED** [review]

**Review by amedvedchuk — APPROVED** [review]

**Pull request merged**